### PR TITLE
(feat) adding git commit rake task

### PR DESCRIPTION
**Pull request by tphoney**

### CODE REVIEW

File: Rakefile [hunk 1]
```diff
@@ -1,37 +1,50 @@
...
 end
 
-desc 'Clone managed modules'
-task :clone_managed_modules do
-  PdkSync::main(steps: [:clone])
-end
+namespace :pdk do 
+  desc 'Runs PDK convert against modules'
+  task :pdk_convert do
+    PdkSync::main(steps: [:pdk_convert])
```
Comment (HelenCampbell) [inline Rakefile]: This is probably extended functionality, but should we be able to take arguments in for the sake of 'template_url'? Would be useful for people to be able to determine their own template.

File: lib/pdksync.rb [hunk 2]
```diff
@@ -58,8 +58,17 @@ def self.main(steps: [:clone, :pdksync, :createpr], args: nil)
     @module_names = return_modules
     # The current directory is saved for cleanup purposes
     @main_path = Dir.pwd
-    exit 'No command passed' if args.nil? && (steps.first == :run_a_command)
-    puts "Running '#{args}'" if !args.nil? && (steps.first == :run_a_command)
+
+    # validation run_a_command
+    if steps.first == :run_a_command
+      raise 'No command passed' if args.nil?
+      puts "Command '#{args}'"
```
Comment (HelenCampbell) [inline lib/pdksync.rb]: I think you should change this from 'Command', as there technically wouldn't be any, to something along the lines of "PDKsync requires an argument to run".

File: lib/pdksync.rb [hunk 3]
```diff
@@ -92,8 +101,13 @@ def self.main(steps: [:clone, :pdksync, :createpr], args: nil)
         print 'updated, '
         next unless pdk_update(@output_path).exitstatus.zero?
       end
-      if steps.include?(:create_pr)
-        print 'PR, '
+      if steps.include?(:create_commit)
+        print 'commit created, '
```
Comment (HelenCampbell) [inline lib/pdksync.rb]: I'd move this output to below the command execution incase it breaks, it'd be confusing for it to tell you the commit is created before it actually is.

**Comment by HelenCampbell** [comment]
Tests on travis failing due to ssh key requirement - testing locally to get this PR merged and all tests have passed:
```
➜  pdksync git:(custom_commit) ✗ bundle exec rspec spec

PdkSync
  main method
puppetlabs-testing, delete module directory, cloned, done.
    runs clone sucessfully
puppetlabs-testing, converted, done.
    runs pdk convert, and files have changed
    raise when running a command with no argument
Command 'touch cat.meow'
puppetlabs-testing, running a command, done.
    runs a command "touch cat.meow"
    raise when create_commit with no arguments
Commit branch_name=temp_branch commit_message=kittens in mittens
puppetlabs-testing, Creating the following branch: pdksync_temp_branch.
All files have been staged.
Creating the following commit: kittens in mittens.
commit created, done.
    create_commit runs, and contains the "kittens in mittens"
  Create a filespace and clone a repo
    has a filespace
    has cloned the repo
  run pdk_update
Creating the following branch: pdksync_1534243644.
    has created a branch
PDK update has run.
    has created a report
All files have been staged.
    has staged files
Creating the following commit: pdksync_1534243644.
    has committed files

Finished in 32.21 seconds (files took 0.56089 seconds to load)
12 examples, 0 failures
```

**Comment by pmcmaw** [comment]
Im happy for this to be merged, some great work and being able to run everything in steps is fantastic. Love love love this.

**Comment by pmcmaw** [comment]
Tests are passing locally on my machine, it is agreed this functionality should go in. This is related to ssh keys. All additional functionality works and has been tested manually also.

Merging.

**Pull request merged**